curriculum/challenges/english/15-javascript-algorithms-and-data-structures-22/learn-basic-string-and-array-methods-by-building-a-music-player/653639d63a45a077333312c8.md

```diff
@@ -0,0 +1,644 @@
+---
+id: 653639d63a45a077333312c8
+title: Step 23
+challengeType: 0
+dashedName: step-23
+---
+
+# --description--
+
+Before playing the song, you need to make sure it starts from the beginning. This can be achieved by the use of the `currentTime` property of the `audio` object. 
+
+Add an `if` statement to check whether the `userData?.currentSong` is `null` or if the `id` of the current song `userData?.currentSong.id` is different from the `id` of the selected song `song.id`. If either of these conditions is true, set the `currentTime` property of the `audio` object to `0`.
+
+
+# --hints--
+
+Test 1
+
+```js
+
+```
```

Comment:
The tests for this step:

You should create an `if` statement with the condition `userData?.currentSong === null || userData?.currentSong.id !== song.id`.

```js
assert.match(code, /if\s*\(userData\?\.currentSong\s*===\s*null\s*\|\|\s*userData\?\.currentSong\.id\s*!==\s*song\.id\)\s*\{\s*/)
```

You should set `audio.currentTime` to `0` inside your `if` block.

```js
assert.match(code, /if\s*\(userData\?\.currentSong\s*===\s*null\s*\|\|\s*userData\?\.currentSong\.id\s*!==\s*song\.id\)\s*\{\s*audio\.currentTime\s*=\s*0;?\s*\}/)
```